File: example/src/com/appbrain/example/BannerActivity.java
package com.appbrain.example;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.AdapterView.OnItemSelectedListener;
import android.widget.LinearLayout.LayoutParams;
import android.widget.ArrayAdapter;
import android.widget.FrameLayout;
import android.widget.LinearLayout;
import android.widget.Spinner;

import com.appbrain.AppBrainBanner;

public class BannerActivity extends Activity {
    private static final String[] COLOR_NAMES = {
        "0, Dark and blue",
        "1, Light and blue",
        "2, Light design",
        "3, Dark and blue",
        "4, Blue and orange",
        "5, White and green",
        "6, Candy",
        "7, Sahara",
        "8, Red",
        "9, Green",
        "10, Blue",
        "11, Purple",
        "12, Light grey",
        "13: Light b&w",
    };
    private static final String[] DESIGN_NAMES = {
        "Simple 1",
        "Simple 2",
        "Arrows",
        "Ellipse"
    };
    
    private static final int TITLE_COUNT = 4, BUTTON_TEXT_COUNT = 3;
    
    private Spinner title, button, design, color;
    private FrameLayout container;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle("AppLift banner playground");
        LinearLayout mainLayout = new LinearLayout(this);
        mainLayout.setOrientation(LinearLayout.VERTICAL);
        
        title = new Spinner(this);
        String[] titles = new String[TITLE_COUNT];
        for (int i = 0; i < titles.length; i++) {
            titles[i] = "Title " + i;
        }
        title.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_dropdown_item_1line, titles));
        
        button = new Spinner(this);
        String[] buttons = new String[BUTTON_TEXT_COUNT];
        for (int i = 0; i < buttons.length; i++) {
            buttons[i] = "Button " + i;
        }        
        button.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_dropdown_item_1line, buttons));
        
        design = new Spinner(this);
        design.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_dropdown_item_1line, DESIGN_NAMES));
        color = new Spinner(this);
        color.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_dropdown_item_1line, COLOR_NAMES));
        
        container = new FrameLayout(this);
        
        OnItemSelectedListener listener = new OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> arg0, View arg1, int arg2, long arg3) {
                loadAd();
            }

            @Override
            public void onNothingSelected(AdapterView<?> arg0) {
            }
        };
        
        title.setSelection(0);
        button.setSelection(0);
        design.setSelection(0);
        color.setSelection(0);
        
        mainLayout.addView(title);
        mainLayout.addView(button);
        mainLayout.addView(design);
        mainLayout.addView(color);
        
        View filler = new View(this);
        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(LayoutParams.FILL_PARENT, 0);
        params.weight = 1.0f;
        mainLayout.addView(filler, params);
        
        mainLayout.addView(container);
        
        loadAd();
        
        setContentView(mainLayout);
        
        title.setOnItemSelectedListener(listener);
        button.setOnItemSelectedListener(listener);
        design.setOnItemSelectedListener(listener);
        color.setOnItemSelectedListener(listener);
        
    }

    private void loadAd() {
        container.removeAllViews();
        AppBrainBanner banner = new AppBrainBanner(BannerActivity.this);
        banner.setTitleIndex(title.getSelectedItemPosition());
        banner.setButtonTextIndex(button.getSelectedItemPosition());
        banner.setDesign(design.getSelectedItemPosition());
        banner.setColors(color.getSelectedItemPosition());
        container.addView(banner);
        banner.requestAd();
    }
}

File: example-compose/src/com/appbrain/example/BannerActivity.kt
package com.appbrain.example

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.key
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.compose.ui.viewinterop.AndroidView

import com.appbrain.AdId
import com.appbrain.AppBrainBanner
import com.appbrain.BannerListener

class BannerActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setExampleContent {
            BannerScreen()
        }
    }

    @Composable
    private fun BannerScreen() {
        val context = LocalContext.current

        var titleIndex by remember { mutableIntStateOf(0) }
        var buttonIndex by remember { mutableIntStateOf(0) }
        var designIndex by remember { mutableIntStateOf(0) }
        var colorIndex by remember { mutableIntStateOf(0) }

        val titles = remember { List(TITLE_COUNT) { "Title $it" } }
        val buttons = remember { List(BUTTON_TEXT_COUNT) { "Button $it" } }

        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(8.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            Text("AppBrain banner playground", style = MaterialTheme.typography.titleLarge)

            Selector("Title", titles, titleIndex) { titleIndex = it }
            Selector("Button", buttons, buttonIndex) { buttonIndex = it }
            Selector("Design", DESIGN_NAMES, designIndex) { designIndex = it }
            Selector("Colors", COLOR_NAMES, colorIndex) { colorIndex = it }

            Box(modifier = Modifier.weight(1f))

            // Recreate the banner whenever a selection changes.
            key(titleIndex, buttonIndex, designIndex, colorIndex) {
                AndroidView(
                    modifier = Modifier.fillMaxWidth(),
                    factory = { ctx ->
                        AppBrainBanner(ctx).apply {
                            setBannerListener(object : BannerListener {
                                override fun onAdRequestDone(hasAd: Boolean) {
                                    // Ignore
                                }

                                override fun onClick() {
                                    context.toast("Banner clicked!")
                                }
                            })
                            setAdId(AdId.DEFAULT)
                            setSize(AppBrainBanner.BannerSize.RESPONSIVE)
                            setTitleIndex(titleIndex)
                            setButtonTextIndex(buttonIndex)
                            setDesign(designIndex)
                            setColors(colorIndex)
                        }
                    }
                )
            }
        }
    }

    @Composable
    private fun Selector(
        label: String,
        options: List<String>,
        selectedIndex: Int,
        onSelected: (Int) -> Unit
    ) {
        var expanded by remember { mutableStateOf(false) }
        Box {
            OutlinedButton(
                onClick = { expanded = true },
                modifier = Modifier.fillMaxWidth()
            ) {
                Text("$label: ${options[selectedIndex]}")
            }
            DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
                options.forEachIndexed { index, option ->
                    DropdownMenuItem(
                        text = { Text(option) },
                        onClick = {
                            onSelected(index)
                            expanded = false
                        })
                }
            }
        }
    }

    companion object {
        private val COLOR_NAMES = listOf(
            "0, Dark and blue",
            "1, Light and blue",
            "2, Light design",
            "3, Dark and blue",
            "4, Blue and orange",
            "5, White and green",
            "6, Candy",
            "7, Sahara",
            "8, Red",
            "9, Green",
            "10, Blue",
            "11, Purple",
            "12, Light grey",
            "13: Light b&w")

        private val DESIGN_NAMES = listOf(
            "Simple 1",
            "Simple 2",
            "Arrows",
            "Ellipse")

        private const val TITLE_COUNT = 4
        private const val BUTTON_TEXT_COUNT = 3
    }
}
